### PR TITLE
Automate running terraform docs 66

### LINKED ISSUE #66: Automate running Terraform Docs commands in CI/CD
### Overview
We need to automate running Terraform Docs commands in our CI/CD pipeline whenever .tf files are changed inside of a Terraform directory, so that ?????

#### Details
For instance, if a .tf file changes inside of terraform-incubator/access-the-data (or any of its subdirectories), README.md file(s) need to be updated using Terraform Docs. 

### Action Items
- [x] Read over the Terraform Docs documentation to understand how the templates are used to generate README.md files
- [x] Read over the Terraform Docs GitHub Action documentation to understand how to integrate the necessary steps into our CI/CD pipeline
- [x] Make changes to [terraform-plan.yml](https://github.com/hackforla/incubator/blob/main/.github/workflows/terraform-plan.yml). This should be able to be one job added as shown in the Terraform Docs GitHub Action

### Resources/Instructions
[Terraform Docs documentation](https://terraform-docs.io/user-guide/introduction/)
[Terraform Docs GitHub Action](https://github.com/terraform-docs/gh-actions)

### PR DESCRIPTION
**Pull request by awlFCCamp**
Fixes #66 

We need to automate running Terraform Docs  whenever a .tf file is changed inside of a Terraform directory. I have changed the region in terraform-incubator/people-depot/dev/main.tf to trigger. I will change back later.

### CODE REVIEW

**Comment by awlFCCamp** [comment]
@chelseybeck @freaky4wrld This is my first draft. Please take a look at it and give feedback for revision. Thank you.

**Comment by freaky4wrld** [comment]
@awlFCCamp I'm not that much experienced with terraform, but I can definitely review it if you provide me details what changes you've made. Or if @chelseybeck tells me where to start from :)

**Comment by freaky4wrld** [comment]
Hey @awlFCCamp I think the idea is to make changes to the [terraform-plan.yml](https://github.com/hackforla/incubator/blob/main/.github/workflows/terraform-plan.yml) , and add the new GHA triggers to it... the file `terraform-docs.yml` looks fine to me. Moreover it can only be observed more when you have your test-environment ready for it!!

Though I'm not sure how you made changes to `README.md` of different projects, did you use `terraform-docs` to do it manually ?

**Comment by freaky4wrld** [comment]
> @freaky4wrld Thank you! Chelsey suggested I put changes into a separate file for testing. I will ask her if I need to revert. I made all the changes through the terraform-docs.yml file. I set up testing in my branch, can you elaborate more on "when you have your test-environment ready for it!!" Sorry, My first GitHub action, I am excited but also a bit unsure.

I was asking if you have tested the GHA in your local repo, if it works fine in the local repo.... it will work here as well, this is the link to the guide in case you need it https://github.com/hackforla/website/wiki/Hack-for-LA's-GitHub-Actions

**Comment by awlFCCamp** [comment]
@freaky4wrld Thank you. It worked last week https://github.com/awlFCCamp/incubator/actions/runs/8725012627, but it failed here. I will look into it later today.

File: terraform-incubator/people-depot/dev/main.tf [hunk 1]
```diff
@@ -9,7 +9,7 @@ terraform {
 }
 
 provider "aws" {
-  region = "us-west-2"
+  region = "us-east-2"
```
Comment (chelseybeck) [inline terraform-incubator/people-depot/dev/main.tf:12]: curious about this change. this is to a main.tf file, not a readme...and the region should prob be us-west-2

File: .github/workflows/terraform-docs.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+
+      - name: Generate Terraform docs terraform-modules
+        uses: terraform-docs/gh-actions@v1.1.0
+        with:
+          working-dir: terraform-modules
```
Comment (chelseybeck) [inline .github/workflows/terraform-docs.yml:19]: we don't want to run terraform docs in directories that don't have a tf file. this is one of those instances, but we do want to run it inside of the other directories in terraform-modules. i wonder if there's a way to refactor this a little where we get a list of the directories and then loop through it

Comment (awlFCCamp) [reply .github/workflows/terraform-docs.yml:19]: Thank you, Chelsey! Great idea, will look into it.

**Comment by awlFCCamp** [comment]
will make another one from a different branch

**Pull request closed without merging**